assets/src/application/ui/layout/entityList/filterBar/subscription/useFilterBarService.ts

```diff
@@ -0,0 +1,28 @@
+import { filter } from 'ramda';
+
+import { useSubscriptionService } from '@appServices/subscription';
+import { FilterBarService, FilterBarServiceHook } from './types';
+import { serviceName as service } from './constants';
+
+type FBS = FilterBarService;
+type FBSH = FilterBarServiceHook;
+
+const useFilterBarService: FBSH = (domain) => {
+	const { getSubscriptions: getServiceSubscriptions, ...restServices } = useSubscriptionService({
+		domain,
+		service,
+	});
+
+	const getSubscriptions: FBS['getSubscriptions'] = (options = {}) => {
+		const { listId } = options;
+		const allSubscriptions = getServiceSubscriptions();
+		if (listId) {
+			return filter(({ options }) => listId === options.listId, allSubscriptions);
+		}
+		return allSubscriptions;
+	};
+
+	return { ...restServices, getSubscriptions };
+};
```

Comment:
This is not yet complete. It will change according to our needs for filter bar UI

Reply:
ya how were you thinking of storing UI elements in the registry?

I see three options:

- using more or less the same code that is already in place, but UI elements would simply use a different service identifier, ex:

     - `useSubscriptionService( 'eventEditor', 'entityListFilters' )` for callbacks
vs: 
     - `useSubscriptionService( 'eventEditor', 'entityListFilterBarUI' )` for JSX

OR

- both callbacks and UI elements use the exact same code that is already in place, but define what they are within their options, ex:

```ts
const options = { type: 'UI' }; // vs 'SERVICE'
const elements = getSubscriptions( options );
```

OR

- a parallel system is built but instead of "Services" it handles UI/JSX Elements, ie: `useUiElementSubscription()` or `useComponentSubscription()` 

Again, just throwing out ideas.

- if any of these ideas work for you => GREAT
- if any of these ideas spark a better idea in your mind => GREAT
- if these ideas all stink but you know what to do anyways => GREAT

Reply:
I will check which one works better.